src/lib/blueprint.ts

```diff
@@ -304,8 +304,51 @@ const createRequest = (
     methods: [uppercaseMethod],
     semanticMethod: uppercaseMethod,
     preferredMethod: uppercaseMethod,
-    parameters: createParameters(operation),
+    parameters: createRequestBody(operation),
+  }
+}
+
+const createRequestBody = (operation: OpenapiOperation): Parameter[] => {
+  // This should be done by the createParameters but for some reason it's not
+  // TODO: remove this in favour of using createParameters
+  if (!('requestBody' in operation) || operation.requestBody === undefined) {
+    return []
+  }
+
+  const requestBody = operation.requestBody
+
+  if (
+    requestBody.content?.['application/json']?.schema?.properties === undefined
+  )
+    return []
+
+  const schema = requestBody.content['application/json'].schema
+
+  if (schema.type !== 'object' || schema.properties == null) {
+    return []
   }
+
+  const requiredProperties = schema.required ?? []
+
+  return Object.entries(schema.properties).map(
+    ([name, property]: [string, any]): Parameter & {
+      type: string
+      format: string
+    } => {
+      const parsedProperty = PropertySchema.parse(property)
+
+      return {
+        name,
+        type: parsedProperty.type,
+        format: property.format,
```

Comment:
We need to define these in the blueprint spec. I'm not sure if we need type and format. I guess the main use case for type would be "json type" and format is "semantic meaning". But if we know something is a date or a uuid, then we also know it is a string.

I think at the blueprint level, it's more important to build on the semantics. So we IMO we want

```
export type Property =
  | StringProperty
  | EnumProperty
  | RecordProperty
  | ListProperty
  | ObjectProperty
  | BooleanProperty
  | DatetimeProperty
  | IdProperty
```

but we can still include the low level type info. How about we remove `type` and do

```
interface StringProperty extends BaseProperty {
  format: 'string'
  jsonType: 'string'
}

interface DatetimeProperty extends BaseProperty {
  format: 'datetime'
  jsonType: 'string'
}

// ...
```